src/codeu/chat/server/Model.java

```diff
@@ -140,10 +146,16 @@ public InterestSet getInterestSet(Uuid id){
     LOG.info("CURRENT: "+interestMap.size());
     return interestMap.get(id);
   }
+  
   public void updateInterests(Uuid id, InterestSet intSet){
     LOG.info("BEFORE: "+interestMap.size());
     interestMap.put(id, intSet);
     LOG.info("AFTER: "+interestMap.size());
     LOG.info(interestMap.get(id).toString());
   }
+  
+  public HashMap<Uuid, Byte> getPermissionMap(ConversationHeader c){
+	  return (HashMap<Uuid, Byte>) authority.get(c.title);
```

Comment:
In general, you should avoid casting where possible, because it introduces the possibility of `ClassCastExceptions` into your program, which can be hard to avoid as your code evolves over time.

You actually don't need to do this here - you can just change the the `authority` map to have type `HashMap<String, HashMap<Uuid, Byte>>`. Then `authority.get(...)` will return an object with type `HashMap<Uuid, Byte>`, no casting necessary!